File: Modulo4-ObjectiveC-Lorena/Exercicio16/Exercicio16.m
//
//  main.m
//  Exercicio16-C
//
//  Created by Lorena Rodrigues Bruno on 11/09/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Pessoa.h"
#import "Conta.h"

@implementation Pessoa
@synthesize nome, idade, cpf;
-(id)initDados:(char*)nome idade:(int)idade cpf:(char*)cpf {
    self = [super init];
    self.nome = nome;
    self.idade = idade;
    self.cpf = cpf;
    return self;
}
-(void)imprimeDados {
    printf("Nome: %s\nCPF: %s\nIdade: %i\n", nome, cpf, idade);
}
@end

@implementation Conta
@synthesize cliente, numeroConta, saldo, limite;
-(id)initConta:(char *)numeroConta saldo:(double)saldo limite:(double)limite cliente:(Pessoa *)cliente{
    self = [super init];
    self.numeroConta = numeroConta;
    self.saldo = saldo;
    self.limite = limite;
    self.cliente = cliente;
    return self;
}
-(void)depositar:(int)valorDeposito{
    saldo += valorDeposito;
}
-(BOOL)sacar:(int)valorSaque{
    if(valorSaque > limite || valorSaque > saldo) {
        printf("Você não tem limite/saldo para realizar esta operação\n");
        return NO;
    }
    else {
        saldo -= valorSaque;
        return YES;
    }
}
-(void)mostraSaldo{
    printf("Saldo: %.2lf\n", saldo);
    }

@end

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        Pessoa *cliente1 = [[Pessoa alloc] initDados:"João" idade:45 cpf:"123456789-10"];
        Conta *conta1 = [[Conta alloc] initConta:"12345" saldo:1000.0 limite:5000.0 cliente:cliente1];
        [conta1 sacar:2000.0 ];
        [conta1 depositar:3000.0];
        [conta1 mostraSaldo];
        
        
    }
    return 0;
}

File: Modulo4-Lorena/Exercicio16.swift
//
//  main.swift
//  Exercicio	16
//
//  Created by Lorena Rodrigues Bruno on 28/08/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

import Foundation

class Pessoa {
    var nome = String()
    var idade = Int()
    var CPF = String()
    
    init(nome : String, idade : Int, CPF : String) {
        self.nome = nome
        self.idade = idade
        self.CPF = CPF
    }
    
    func imprimeInformacoes() {
        print("\(nome), de numero de CPF: \(CPF), tem \(idade) anos. ")
    }
}

class Conta {
    var numeroDaConta = String()
    var saldo = Int()
    var limite = Int()
    var pessoa = Pessoa(nome: "Maria", idade: 32, CPF: "000111222-33")
    
    init(numeroDaConta: String, saldo: Int, limite: Int) {
        self.numeroDaConta = numeroDaConta
        self.limite = limite
        self.saldo = saldo
    }
    
    func depositar(valorDoDeposito : Int) {
        saldo += valorDoDeposito
    }
    
    func sacar(valorDoSaque : Int) -> Bool {
        if saldo > valorDoSaque && valorDoSaque < limite {
            saldo -= valorDoSaque
            return true
        } else {
            return false
        }
    }
    
    func verificarSaldo() {
        print("Seu saldo é de R$\(saldo)")
    }
}

//teste
//var verificador : Bool = true
//var conta = Conta(numeroDaConta: "1234", saldo: 5_000, limite: 500)
//conta.depositar(valorDoDeposito: 700)
//verificador = conta.sacar(valorDoSaque: 200)
//if(verificador == false) {
//    print("Operação inválida")
//}
//conta.verificarSaldo()


